numpy/core/src/umath/loops.c.src

```diff
@@ -739,7 +739,8 @@ BOOL_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED
              * with glibc >= 2.12 and memchr can only check for equal 1
              */
             static const npy_bool zero[4096]; /* zero by C standard */
-            npy_uintp i, n = dimensions[0];
+            npy_intp n = dimensions[0];
+            npy_intp i;
 
             for (i = 0; !*op && i < n - (n % sizeof(zero)); i += sizeof(zero)) {
```

Comment:
Isn't this expression a mixed comparison too, since `sizeof` returns an unsigned type?

Reply:
Indeed. In fact, the travis daily errors do not match up with the ones I see either, and apparantly travis has also changed something, because the daily builds are no longer failing. Hmm...

Many of the warnings have been for things that are harmless, but there may be a few that matter. In particular, here in the loops there are places where `steps` is assumed positive in multiplications, which, AFAIK, is not guaranteed. Multiplying a negative times unsigned can cause obvious problems ... 

I'm inclined to put a lower priority on these fixes as the tests are now passing; the issues can be subtle and need thought. However, there are some actual potential errors that we should search out and address, maybe just with some asserts scattered here and there.

Reply:
I note that gcc is not catching this comparison, I wonder why not? Possibly `sizeof(zero)` has been turned into an integer literal somewhere in the compilation process, The information is certainly there.

Reply:
So, gcc has this to say:

>-Wsign-compare
Warn when a comparison between signed and unsigned values could produce an incorrect result when the signed value is converted to unsigned. In C++, this warning is also enabled by -Wall. In C, it is also enabled by -Wextra.

Apparently, it is not the case that it always warns.
>

Reply:
Maybe the compiler knows that `i` is always positive?